TC-18: Ds client sslkeys methods

mdb opened the pull request:
Note that this `mdb:ds-client-sslkeys-methods` is branched from `mdb:ds-client`, for which there is an open PR in PR #18. If/when that is merged, this change can be rebased to only include commit `ec429d3`.

Thanks!

dneuman64 commented on traffic_ops/client/delivery_service.go:
```diff
@@ -161,6 +161,30 @@ type DeliveryServiceServer struct {
+	Country      string                            `json:"country"`
+	State        string                            `json:"state"`
+	Version      string                            `json:"version"`
+	Certificate  DeliveryServiceSSLKeysCertificate `json:"certificate"`
```
I think we want to add Cdn and Deliveryservice fields to this struct...here is a sample output: 

```
{
  "response": {
    "country": "US",
    "cdn": "qa",
    "deliveryservice": "my-ds",
    "org": "Kabletown",
    "version": "1",
    "certificate": {
      "csr": "mycsr",
      "crt": "mycrt",
      "key": "mykey"
    },
    "hostname": "*.my-ds.qa.kabletown.net",
    "key": "my-ds",
    "state": "Colorado",
    "city": "Denver",
    "unit": "CDNENG"
  }
}

```

mdb commented:
@dneuman64 Thanks! I've added the fields, and also updated the documentation. Please let me know if I should fine-tune the documentation language; I'm not very informed on how best to document these fields.

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -770,29 +770,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``cdn``             | string | The CDN entered by the user when generating certs.                                                                                      |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``deliveryservice`` | string | The delivery service entered by the user when generating certs.                                                                         |
```
The xml_id of the delivery service the cert was generated for.  This is not user entered.

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -770,29 +770,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``key``             | string | base64 encoded private key file for delivery service                                                                                    |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``cdn``             | string | The CDN entered by the user when generating certs.                                                                                      |
```
The CDN of the Delivery Service for which the certs were generated.  This is not user entered.

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -770,29 +770,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``organization``    | string | The organization entered by the user when generating certs.  Field is optional and if not provided by the user will not be in response  |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``hostname``        | string | The hostname entered by the user when generating certs.  Field is optional and if not provided by the user will not be in response      |
```
The hostname generated by Traffic Ops that is used as the common name when generating the certificate.  This will be a FQDN for DNS delivery services and a wildcard URL for HTTP delivery services.  This field is not user entered.

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -840,29 +846,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``cdn``             | string | The CDN entered by the user when generating certs.                                                                                      |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``deliveryservice`` | string | The delivery service entered by the user when generating certs.                                                                         |
```
see above

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -840,29 +846,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``organization``    | string | The organization entered by the user when generating certs.  Field is optional and if not provided by the user will not be in response  |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``hostname``        | string | The hostname entered by the user when generating certs.  Field is optional and if not provided by the user will not be in response      |
```
see above

dneuman64 commented on docs/source/development/traffic_ops_api/v12/deliveryservice.rst:
```diff
@@ -840,29 +846,33 @@ SSL Keys
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``key``             | string | base64 encoded private key file for delivery service                                                                                    |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``cdn``             | string | The CDN entered by the user when generating certs.                                                                                      |
```
see above

dneuman64 commented:
@mdb fyi, I merged #18, so this is ready for a rebase in addition to the suggestions I made above.

mdb commented:
@dneuman64 Thanks! I've rebased & pushed your documentation edits.
